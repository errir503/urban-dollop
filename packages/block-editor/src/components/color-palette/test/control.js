/**
 * External dependencies
 */
import { create, act } from 'react-test-renderer';

/**
 * Internal dependencies
 */
import ColorPaletteControl from '../control';

const noop = () => {};

describe( 'ColorPaletteControl', () => {
	it( 'matches the snapshot', async () => {
		let root;

		await act( async () => {
			root = create(
				<ColorPaletteControl
					label="Test Color"
					value="#f00"
					colors={ [ { color: '#f00', name: 'red' } ] }
					disableCustomColors={ false }
					onChange={ noop }
				/>
			);
		} );

		expect( root.toJSON() ).toMatchSnapshot();
	} );
} );
