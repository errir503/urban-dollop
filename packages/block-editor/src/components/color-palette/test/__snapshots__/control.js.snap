// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPaletteControl matches the snapshot 1`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 13px;
}

.emotion-2 {
  margin-bottom: calc(4px * 2);
}

.components-panel__row .emotion-2 {
  margin-bottom: inherit;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-4>*+*:not( marquee ) {
  margin-top: calc(4px * 1);
}

.emotion-4>* {
  min-height: 0;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-6>*+*:not( marquee ) {
  margin-top: calc(4px * 3);
}

.emotion-6>* {
  min-height: 0;
}

<div
  className="components-base-control block-editor-color-gradient-control emotion-0 emotion-1"
>
  <div
    className="components-base-control__field emotion-2 emotion-3"
  >
    <fieldset>
      <div
        className="components-flex components-h-stack components-v-stack emotion-4 emotion-5"
        data-wp-c16t={true}
        data-wp-component="VStack"
      >
        <legend>
          <div
            className="block-editor-color-gradient-control__color-indicator"
          >
            <span
              className="components-base-control__label"
            >
              Test Color
            </span>
          </div>
        </legend>
        <div
          className="components-flex components-h-stack components-v-stack emotion-6 emotion-5"
          data-wp-c16t={true}
          data-wp-component="VStack"
        >
          <div
            className="components-dropdown"
            tabIndex="-1"
          >
            <button
              aria-expanded={false}
              aria-haspopup="true"
              aria-label="Custom color picker"
              className="components-color-palette__custom-color"
              onClick={[Function]}
              style={
                Object {
                  "background": "#f00",
                  "color": "#000",
                }
              }
            >
              #f00
            </button>
          </div>
          <div
            className="components-circular-option-picker"
          >
            <div
              className="components-circular-option-picker__swatches"
            >
              <div
                className="components-circular-option-picker__option-wrapper"
              >
                <button
                  aria-describedby={null}
                  aria-label="Color: red"
                  aria-pressed={true}
                  className="components-button components-circular-option-picker__option is-pressed"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#f00",
                      "color": "#f00",
                    }
                  }
                  type="button"
                />
                <svg
                  aria-hidden={true}
                  fill="#000"
                  focusable={false}
                  height={24}
                  role="img"
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.7 7.1l-6.3 8.5-3.3-2.5-.9 1.2 4.5 3.4L17.9 8z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="components-circular-option-picker__custom-clear-wrapper"
            >
              <button
                aria-describedby={null}
                className="components-button components-circular-option-picker__clear is-secondary is-small"
                onClick={[Function]}
                type="button"
              >
                Clear
              </button>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;
